### ISSUE TITLE
[feature] Support semver with 'v' prefix when calculating package id

### ISSUE THREAD

**Issue by markushedvall** [conan-io/conan#8066]
Where I'm working we have over 100+ conan packages using sematic versioning, but all with a 'v' prefix in their version. For example, a version could be v3.2.1. This still works well when using range versions for requirements as node-semver will clean the version, effectively ignoring the 'v' prefix.

However, the logic used when calculating the package id does not see it as a semver version. So `semver_mode()` will basically work the same as `full_version_mode()`.

My suggestion is to use the clean function in node-semver (https://github.com/npm/node-semver#clean) to clean the version in `model/version.py`:

```diff
             self._cached_list = []
-            tokens = Version.version_pattern.split(tokens[0])
+            cleaned = semver.clean(tokens[0], loose=False) or tokens[0]
+            tokens = Version.version_pattern.split(cleaned)
             for item in tokens:
                 self._cached_list.append(int(item) if item.isdigit() else item)
         return self._cached_list
```

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @markushedvall 

Though this issue kind of seems to make sense, I am afraid it would be massively breaking other users. Every user that have their packages with version ``vX.Y.Z``, which is completely valid and works perfectly, (also they might not even be using version-ranges at all) will suddenly have no binaries in their servers. 

Moreover, the fact that node-semver support it might not be the best reason, as it seems it is some legacy behavior of the previous Semver 1.0.0 specification (https://semver.org/spec/v1.0.0.html)  which has been superseded by 2.0.0 (https://semver.org/spec/v2.0.0.html) which is there since 2013: https://github.com/semver/semver/releases. I wouldn't rely on that behavior at all, as node-semver might change behavior in newer releases, or Conan might switch the semver implementation, this has already been requested by other users (e.g. https://github.com/conan-io/conan/issues/2172), to other implementation (this is likely to happen).

So IMHO this is too risky at the moment, and I would strongly prefer to wait until Conan 2.0, and only after deciding on the versioning scheme and the underlying semver library, we might consider this feature (no guarantee that it will be possible).

**Comment by markushedvall** [comment]
Thanks for the quick reply! That makes perfect sense.

We are currently settings the package versions directly from git tags (with v prefix). I've tried to convince that we should strip the 'v' to get a valid Semver, but so far I've failed. I'll give it another try, referring to this issue :)

**Comment by memsharded** [comment]
Closing this as not planned, Conan 2.0 introduced new version, semver and version-ranges capabilities, but this ``v`` thing was not. The ecosystem has more or less converged to the convention of not using ``v``.
